napari/components/dims.py

```diff
@@ -65,13 +65,33 @@ class Dims(EventedModel):
     """
 
     # fields
-    ndim: int = 2
-    ndisplay: Literal[2, 3] = 2
-    last_used: int = 0
-    range: Tuple[Tuple[float, float, float], ...] = ()
-    current_step: Tuple[int, ...] = ()
-    order: Tuple[int, ...] = ()
-    axis_labels: Tuple[str, ...] = ()
+    ndim: int = Field(default=2, title=trans._("Number of dimensions"))
+    ndisplay: Literal[2, 3] = Field(
+        default=2, title=trans._("Number of displayed dimensions")
+    )
+    last_used: int = Field(default=0, title=trans._("Last-used dimension"))
+    range: Tuple[Tuple[float, float, float], ...] = Field(
+        default_factory=tuple,
+        title=trans._("Range in each dimension"),
+        description=trans._(
+            "(min_val, max_val, step_size) for each dimension."
+        ),
```

Comment:
>  document events in the code itself using pydantic.Field ....
> makes the code itself a little uglier ... but keeps all of the "meaning" in one place, which is nice.

Is there a way we could parse the existing doc string of `Dims` and then add these, or would that not serve any additional purpose. I do worry that it leads to quite a bit of doc string duplication and makes reading the class a bit harder